src/NuGet.Core/NuGet.Packaging/Signing/Utility/SignatureUtility.cs

```diff
@@ -256,21 +256,24 @@ public static IX509CertificateChain GetCertificateChain(PrimarySignature primary
                 includeChain);
         }
 
-        public static bool HasRepositoryCountersignature(PrimarySignature primarySignature)
+        public static bool HasRepositoryCountersignature(Signature signature)
         {
-            if (primarySignature == null)
+            if (signature == null)
             {
-                throw new ArgumentNullException(nameof(primarySignature));
+                throw new ArgumentNullException(nameof(signature));
             }
 
-            var counterSignatures = primarySignature.SignerInfo.CounterSignerInfos;
-
-            foreach (var counterSignature in counterSignatures)
+            if (signature is AuthorPrimarySignature)
```

Comment:
This breaks the method working on "unknown" primary signatures.

I think this is what you want:

    if (signature is RepositoryPrimarySignature)
    {
       return false;
    }

Then continue with existing code:

    var countersignatures = primarySignature.SignerInfo.CounterSignerInfos;
    ...